x/sync/syncmanager.go

```diff
@@ -106,7 +106,7 @@ type StateSyncManager struct {
 }
 
 type StateSyncConfig struct {
-	SyncDB                merkledb.MerkleDB
+	SyncDB                merkledb.SyncableDB
```

Comment:
Now `x/sync` depends on a smaller interface than `MerkleDB`. Yay!

Reply:
Should we move the new interface to the `sync` package?

Reply:
moved to  `x/sync`